source/common/http/conn_manager_impl.h

```diff
@@ -199,6 +215,16 @@ class ConnectionManagerConfig {
   virtual bool useRemoteAddress() PURE;
 
   /**
+   * @return ForwardClientCertType the configuration of how to forward the client certs.
+   */
+  virtual ForwardClientCertType forwardClientCert() PURE;
+
+  /**
+   * @return ClientCertDetailsType the details of the client cert to forward to the next hop.
+   */
+  virtual const std::list<ClientCertDetailsType> setClientCertDetails() PURE;
```

Comment:
It is not mutating anything. This function is directly returning the set_client_cert_details field in the configuration, that indicates which information from the current client cert should be set into the x-forwarded-client-cert header. It returns a list because there are cases you want to set both subject and SAN into the header, and in the future more information may need to be set into the header as well.